.NET/Microsoft.Recognizers.Text.DateTime/Parsers/BaseDateParser.cs

```diff
@@ -627,7 +627,7 @@ private static DateObject ComputeDate(int cardinal, int weekday, int month, int
                 weekday = 7;
             }
 
-            if (weekday < (int)firstDay.DayOfWeek)
+            if (weekday < ((int)firstDay.DayOfWeek == 0 ? 7 : (int)firstDay.DayOfWeek))
```

Comment:
Please assign the value to a variable and use it in the if condition.
Also, please add test cases, to show that the bug was really fixed and that other cases that hit this code are not broken.